app/scripts/models/reliers/oauth.js

```diff
@@ -260,12 +311,39 @@ define(function (require, exports, module) {
   }
 
   function scopeStrToArray(scopes) {
-    return typeof scopes === 'string' ? _.uniq(scopes.split(/\s+/g)) : scopes;
+    if (! _.isString) {
+      return [];
+    }
+
+    var trimmedScopes = scopes.trim();
```

Comment:
a scope of `''` would create an array with a single scope, ``. That's not what we wanted.